src/_rules/layout.js

```diff
@@ -17,3 +19,18 @@ export const overflows = [
       (overflowValues.includes(v) ? { [`overflow-${d}`]: v } : undefined),
   ],
 ];
+
+export const columns = [
+  [
+    /^columns-([1-9]|1[0-2])$/,
+    ([, d]) => ({ 'columns': h.number(d) }),
+    { autocomplete: 'columns-<num>' },
+  ],
+  ['columns-auto', { 'columns': 'auto' }],
+  // matching arbitrary values
+  [
+    /^columns-\[(\d*)?]$/,
```

Comment:
Should we allow empty brackets here? Or should it be changed to:
```suggestion
    /^columns-\[(\d+)]$/,
```

Reply:
no you are right! We probably don't want to allow empty brackets. 😄 I'll change it!